src/inmanta/deploy/state.py

```diff
@@ -125,6 +181,73 @@ def reset(self) -> None:
         self.resource_state.clear()
         self.types_per_agent.clear()
 
+    def block_resource(self, resource: ResourceIdStr, details: ResourceDetails, transient: bool) -> None:
+        """
+        Mark the given resource as blocked, i.e. it's not deployable.
```

Comment:
This method also updates the resource details / intent. I think that should be mentioned.